The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/27075839983/job/79913034558) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/79913034558) [(plain)](https://github.com/rust-lang/rust/commit/e673d20363507f1576ec458cb8d42754bd8230b2/checks/79913034558/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] tests/ui/proc-macro/test.rs stdout ----

error: auxiliary build of /checkout/tests/ui/proc-macro/auxiliary/api/proc_macro_api_tests.rs failed to compile: 
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/proc-macro/auxiliary/api/proc_macro_api_tests.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/proc-macro/test/auxiliary" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021" "--crate-type" "proc-macro" "--extern" "proc_macro" "-L" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/proc-macro/test/auxiliary"
stdout: none
--- stderr -------------------------------
error[E0658]: `cfg(target_has_reliable_f16)` is experimental and subject to change
##[error]  --> /checkout/tests/ui/proc-macro/auxiliary/api/literal.rs:105:13
   |
LL |     if cfg!(target_has_reliable_f16) {
   |             ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(cfg_target_has_reliable_f16_f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: `cfg(target_has_reliable_f16)` is experimental and subject to change
##[error]  --> /checkout/tests/ui/proc-macro/auxiliary/api/literal.rs:111:13
   |
LL |     if cfg!(target_has_reliable_f16) {
   |             ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(cfg_target_has_reliable_f16_f128)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

warning: unexpected `cfg` condition name: `target_has_reliable_f16`
##[warning]  --> /checkout/tests/ui/proc-macro/auxiliary/api/literal.rs:105:13
   |
LL |     if cfg!(target_has_reliable_f16) {
```

</details>

--- a/tests/ui/proc-macro/auxiliary/api/literal.rs
+++ b/tests/ui/proc-macro/auxiliary/api/literal.rs
@@ -1,7 +1,5 @@
 // ignore-tidy-linelength
 
-#![feature(f16)]
-
 use proc_macro::Literal;
 
 pub fn test() {
@@ -102,9 +100,15 @@ fn test_literal_value() {
 
     assert!(Literal::f32_suffixed(9.).u8_value().is_err());
     assert!(Literal::f32_suffixed(10.).f64_value().is_err());
-    assert!(Literal::f32_suffixed(11.).f16_value().is_err());
+    #[cfg(target_has_reliable_f16)]
+    {
+        assert!(Literal::f32_suffixed(11.).f16_value().is_err());
+    }
     assert_eq!(Literal::f32_suffixed(12.).f32_value().map(|f| f.to_string()), Ok("12".to_string()));
     assert_eq!(Literal::f32_unsuffixed(13.).f32_value().map(|f| f.to_string()), Ok("13".to_string()));
     assert_eq!(Literal::f32_unsuffixed(14.).f64_value().map(|f| f.to_string()), Ok("14".to_string()));
-    assert_eq!(Literal::f32_unsuffixed(15.).f16_value().map(|f| f.to_string()), Ok("15".to_string()));
+    #[cfg(target_has_reliable_f16)]
+    {
+        assert_eq!(Literal::f32_unsuffixed(15.).f16_value().map(|f| f.to_string()), Ok("15".to_string()));
+    }
 }

--- a/tests/ui/proc-macro/auxiliary/api/proc_macro_api_tests.rs
+++ b/tests/ui/proc-macro/auxiliary/api/proc_macro_api_tests.rs
@@ -2,6 +2,8 @@
 
 #![feature(proc_macro_span)]
 #![feature(proc_macro_value)]
+#![feature(f16)]
+#![feature(cfg_target_has_reliable_f16_f128)]
 #![deny(dead_code)] // catch if a test function is never called
 
 extern crate proc_macro;

